src/core/model/Stroke.cpp

```diff
@@ -375,6 +446,153 @@ auto Stroke::intersects(double x, double y, double halfEraserSize, double* gap)
     return false;
 }
 
+auto Stroke::intersectWithPaddedBox(const PaddedBox& box) const -> IntersectionParametersContainer {
+    auto pointCount = this->points.size();
+    if (pointCount < 2) {
+        if (pointCount == 1 && this->points.back().isInside(box.getInnerRectangle())) {
+            IntersectionParametersContainer result;
+            result.emplace_back(0U, 0.0);
+            result.emplace_back(0U, 0.0);
+            return result;
+        }
+        return IntersectionParametersContainer();
+    }
+    return this->intersectWithPaddedBox(box, 0, pointCount - 2);
+}
+
+auto Stroke::intersectWithPaddedBox(const PaddedBox& box, size_t firstIndex, size_t lastIndex) const
```

Comment:
Can you split this up for me, either into local lambdas or static (file local) functions, with good explaining names 😅 .. 

For a reviewer, this function is horrible, since it has many branches high level of nesting, etc. also it is most likely, that a bug is hiding there, but neither the implementor nor any reviewer will find it.

Therefore, many people defined some metrics to prevent that:
 - e.g. functions should not be longer than 20 lines (of code),
 - reduce multiline comments
 - explain why something must or must not happen, not what.
 - and last but not least: [clang-tidy-check](https://clang.llvm.org/extra/clang-tidy/checks/readability-function-cognitive-complexity.html) [paper cognitive-complexity](https://www.sonarsource.com/docs/CognitiveComplexity.pdf)

Reply:
for example: ` * Initialize with the half tangent at the first point ` can be a function with exact that name.

Reply:
Let me try :-)

Reply:
I made an attempt. You can have a look whenever you have time